### PR TITLE
Initial version of Daxko SSO for Gated Content

### PR DESCRIPTION
**Pull request by anpolimus**

### CODE REVIEW

File: modules/openy_gc_auth/modules/openy_gc_auth_daxko_sso/src/Controller/DaxkoLinkController.php [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+        ]
+      );
+    }
+
+    $backlinkUrl = $request->getSchemeAndHttpHost() . $plugin_config->get('redirect_url');
+    $token = $this->daxkoClient->getUserAccessToken($code, $backlinkUrl);
+    // Based on user token, get logged user data.
+    $userData = $this->daxkoClient->getMyInfo($token);
+
+    $userDetails = $this->daxkoClient->getRequest('members/' . $userData->member_id);
```
Comment (retif) [inline modules/openy_gc_auth/modules/openy_gc_auth_daxko_sso/src/Controller/DaxkoLinkController.php:116]: http call may be wrapped into try-catch to return json error if request is failed

Comment (anpolimus) [reply modules/openy_gc_auth/modules/openy_gc_auth_daxko_sso/src/Controller/DaxkoLinkController.php:122]: Try catch exists at the client side.

File: js/gated-content/src/store/modules/auth/daxkosso.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  },
+  actions: {
+    async daxkossoAuthorize(context) {
+      if (!context.getters.isLoggedIn) {
+        const urlParams = new URLSearchParams(window.location.search);
+        const state = urlParams.get('state');
+        const code = urlParams.get('code');
+
+        if ((state !== null) && (code !== null)) {
+          const apiCheckEndPoint = `/openy-gc-auth-daxko-sso/back-redirect?state=${state}&code=${code}`;
```
Comment (hamrant) [inline js/gated-content/src/store/modules/auth/daxkosso.js:15]: Maybe make sense to move `/openy-gc-auth-daxko-sso/back-redirect`to the plugin config. In this case, developers can create custom routes with additional logic and override this on backend.

File: modules/openy_gc_auth/modules/openy_gc_auth_daxko_sso/config/openy_gc_auth.provider.daxko_sso.yml [hunk 3]
```diff
@@ -0,0 +1 @@
+redirect_url: '/gated_content'
```
Comment (hamrant) [inline modules/openy_gc_auth/modules/openy_gc_auth_daxko_sso/config/openy_gc_auth.provider.daxko_sso.yml:1]: Can we use `/gated-content`?

File: modules/openy_gc_auth/modules/openy_gc_auth_daxko_sso/openy_gc_auth_daxko_sso.info.yml [hunk 4]
```diff
@@ -0,0 +1,9 @@
+name: Daxko Gated Content integration
+type: module
+description: 'Daxko integration for Gated content'
+package: Open Y
```
Comment (hamrant) [inline modules/openy_gc_auth/modules/openy_gc_auth_daxko_sso/openy_gc_auth_daxko_sso.info.yml:4]: Please use `Open Y Gated Content` package

File: modules/openy_gc_auth/modules/openy_gc_auth_daxko_sso/openy_gc_auth_daxko_sso.module [hunk 5]
```diff
@@ -0,0 +1,6 @@
+<?php
```
Comment (hamrant) [inline modules/openy_gc_auth/modules/openy_gc_auth_daxko_sso/openy_gc_auth_daxko_sso.module:1]: Let's remove this file if there no any hooks here

File: modules/openy_gc_auth/modules/openy_gc_auth_daxko_sso/config/openy_gc_auth.provider.daxko_sso.yml (hunk 3 above)
Comment (anpolimus) [reply modules/openy_gc_auth/modules/openy_gc_auth_daxko_sso/config/openy_gc_auth.provider.daxko_sso.yml:1]: That is default link to the gated content node app. that is how it's configured by default in demo content. I've just used it.

**Pull request merged**